src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1019,3 +1019,182 @@ the probability mass function of $z_i$ in the posterior.   These may
+$X$, and discrete parameters, $Y$. We will also assume that $Y$ can only take
+finitely many values, as is the case for all the models described in this 
+chapter of the User's Guide. In order to simplify notation, we suppress any
+conditioning on data in this section, except where specified. As with all 
+Bayesian analyses, however, all inferences using models with marginalized
+parameters are made conditional on the observed data. 
+
+### Estimating expectations {-}
+
+When making Bayesian inferences, we are generally interested in estimating some 
+quantity of interest $\mu = \textsf{E}(g(X, Y))$ where $g(\cdot)$ is an arbitrary 
+function. For example, we may be interested in $\textsf{E}(X)$, the posterior 
+mean of the continuous parameters, where we would take $g(x, y) = x$. In this
+context, by *estimating*, we mean estimating in the *Monte Carlo* sense, which
+refers to calculating a series of low-dimensional summary statistics of the
+posterior distribution of the quantity of interest by processing the numerical
+values from a large number of posterior draws. This is in contrast to the usual
+frequentist approach to estimation that is concerned with bias and sampling
+variation of estimators that generate point estimates. When doing Bayesian 
+analyses, the Monte Carlo estimation procedure is often trivial to perform, 
+but more care is required when working with marginalized posteriors as we 
+describe in this section.
+
+If both $X$ and $Y$ are continuous, we would use Stan to sample $M$ draws
+from the joint posterior $p_{X, Y}(x, y)$ and then estimate $\mu$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M {g(x_i, y_i)}.
+$$
+
+If $Y$ is discrete, however, we cannot use Stan to sample from the joint 
+posterior because Stan cannot compute the derivatives it needs for the 
+discrete parameters. 
+
+Instead, as outlined in the previous sections, we first marginalize out $Y$ 
+from the joint posterior to give the marginalized posterior $p_X(x)$. This
+marginalized posterior can then be implemented in Stan as usual, and Stan will 
+give samples $\{x_i\}_{i = 1}^M$ from the marginalized posterior. 
+
+How, using only the samples $\{x_i\}_{i = 1}^M$, can we estimate 
+$\textsf{E}(g(X, Y))$? We can use a conditional estimator. That is, for each
+function $g$ of interest we can compute
+
+$$
+h(X) = \textsf{E}(g(X, Y) \mid X)
+$$
+
+and then estimate $\textsf{E}(g(X, Y))$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M h(x_i).
+$$
+
+This estimator is justified by the fact that
+
+$$
+\textsf{E}(h(X)) 
+= \textsf{E}(\textsf{E}(g(X, Y) \mid X)) 
+= \textsf{E}(g(X, Y)) = \mu,
+$$
+
+which uses the [law of total expectation](https://en.wikipedia.org/wiki/Law_of_total_expectation). Using 
+this marginalized estimator gives us a way to estimate the expectation of any
+function $g$ for any combinations of discrete or continuous parameters in the 
+model. However, it presents a new challenge: evaluating the conditional 
```

Comment:
Nitpicky personal opinion here.  I think it's potentially confusing provide an "answer" to "How can we estimate 
$\textsf{E}(g(X, Y))$?", only to arrive back at the question of "How can we evaluate the conditional expectation?"

I'd suggest collapsing the writing here to say that we use the law of total expectation to arrive at a new expression, which we'll now figure out how to evaluate.

Reply:
I'm OK with it as presented. 

At a high level, I tend to think of the marginalized form of the estimate in two steps,

```
E[f(Theta, Z)]
  = 1/M SUM_{m in 1:M} E[f(theta(m), Z)]
  = 1/M SUM_{m in 1:M} 1/N SUM_{z in Z} Pr(z | theta(m)) f(theta(m), z),
```

where `Theta` is a continuous parameter and `Z` a discrete one, `theta(m) ~ p(theta)` for `m in 1:M`, and where `z in Z` ranges over all legal values of the random variable `Z`.